Update the apollo cache

mattwr18 opened the pull request:
## 🍰 Pullrequest
<!-- Describe the Pullrequest. Use Screenshots if possible. -->

### Issues
<!-- Which Issues does this fix, which are related?

- relates #XXX
-->
- fixes #1391

mattwr18 commented on webapp/components/Comment.vue:
```diff
@@ -140,24 +141,33 @@ export default {
+          this.$emit('deleteComment')
+        })
+        .catch(err => {
+          this.$toast.error(err.message)
```
@Tirokk, I originally tried using `update` with `async/await`, but I couldn't the syntax correct.
I just changed it to using a promise cause I have examples in the code base of it working... we could refactor it, but I'm not sure if there is a need, just to let you know why I refactored it so.

appinteractive replied:
Just add `async` to the outer method, put a `await` before the request and wrap that inside the try/catch

appinteractive commented on webapp/components/Comment.vue:
```diff
@@ -140,24 +141,33 @@ export default {
+        .mutate({
+          mutation: CommentMutations(this.$i18n).DeleteComment,
           variables: { id: this.comment.id },
+          update: store => {
```
`update: async store => {`

appinteractive commented on webapp/components/Comment.vue:
```diff
@@ -140,24 +141,33 @@ export default {
+            if (index !== -1) {
+              data.Post[0].comments.splice(index, 1)
+            }
+            store.writeQuery({ query: PostQuery(this.$i18n), data })
```
`await store.writeQuery(`

appinteractive commented on webapp/components/Comment.vue:
```diff
@@ -140,24 +141,33 @@ export default {
+          mutation: CommentMutations(this.$i18n).DeleteComment,
           variables: { id: this.comment.id },
+          update: store => {
+            const data = store.readQuery({
```
`const data = await store.readQuery(`

mattwr18 commented:
hey thanks @appinteractive... the issue was that I was trying to put the update outside of the closing `}`, but inside the closing `)`... when I moved it inside both, it worked fine. I added the `async/awaits` to the update, readQuery, and writeQuery, as you suggested. I have a couple questions, if/when you have time to answer them.
1) is there any benefit to using `try/catch(async/await)` over `promises/then` syntax? I feel like it's just a style thing, but would love to know if there are any benefits/downsides to each.
2) the docs don't use `async/await` for updates or `readQuery`/`writeQuery`, is that cause the examples are in smart queries, or some other reason you know of?
